sim/signing/fake.go

```diff
@@ -36,10 +32,8 @@ func (s *FakeBackend) Sign(signer gpbft.PubKey, msg []byte) ([]byte, error) {
 }
 
 func (s *FakeBackend) generateSignature(signer gpbft.PubKey, msg []byte) ([]byte, error) {
-	priv, known := s.keyPairs[string(signer)]
-	if !known {
-		return nil, errors.New("unknown signer")
```

Comment:
I can parse and compare with `i`, but that would require testing to produce a public key from nowhere. Are we concerned about that?

Reply:
There is a signing interface in sim/singning package that defines the GenerateKey reciever. 

I think the question worth thinking about is: what should happen here in production code if a given public key is unknown?

Whatever the answer to that question is should be reproducible by the fake signer for testing purposes. Otherwise, fake signer would miss a whole class of behaviour in tests.

Reply:
Then, we should separate the fake sigher by the simulated participant.